enumerable_module.rb

```diff
@@ -0,0 +1,129 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+# rubocop:disable Metrics/ModuleLength
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    to_a.length.times do |i|
+      yield to_a[i]
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    to_a.length.times do |i|
+      yield to_a[i], i
+    end
+  end
```

Comment:
- [ ] `my_each_with_index` is not working as the original `each_with_index` method, also is not passing all test cases.

### `each_with_index` is an Enumerable that has these characteristics: 
- it returns the array/range/hash itself if a block is given.

```
[1, 2, 3].each_with_index { |num, idx| num }
# [1, 2, 3]
(1..3).each { |num, idx| num * idx }
# 1..3
````

[Enumerator#each](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-each_with_index)